### PR TITLE
DOC: Add regression example, FAQ, and clean up other docs

### PR DESCRIPTION
**Pull request by richford**

### CODE REVIEW

**Comment by arokem** [comment]
Looks great!

I have a request: could you please change the funding source on the front page to be the same as in the README? 

And a question: why is performance in the regression prediction perfect? The data has noise in it and you are doing cross-validation, so I would expect it to be good (and certainly better than Lasso), but not 0 error. Or am I missing something?

**Comment by richford** [comment]
For the regression example, I'm surprised by the perfect fit as well. But I'm training and scoring on the same data. I'm using a CV estimator but that's the "interior CV" loop to find the right hyperparameters. I'll push a change to the example to use `train_test_split` and evaluate performance on a held out test set.

**Comment by richford** [comment]
Updated the example to use train/test split. Lasso now outperforms SGL slightly on the training data. But SGL outperforms lasso on the test data.

File: doc/index.rst [hunk 1]
```diff
@@ -36,27 +36,29 @@ group-lasso and the lasso, with :math:`\alpha=0` giving the group lasso fit
 and :math:`\alpha=1` yielding the lasso fit. The hyperparameter
 :math:`\lambda` controls the strength of the regularization.
 
+.. toctree::
+   :hidden:
+   :titlesonly:
+
+   Home <self>
```
Comment (arokem) [inline doc/index.rst:43]: Why is this here? AFAICT it just adds a link from the middle of this page to the top of the page (?)

**Comment by arokem** [comment]
Other than that one comment, I think that this is ready (?)

File: doc/index.rst (hunk 1 above)
Comment (richford) [reply doc/index.rst:43]: It adds a "Home" link to the sidebar. I believe there are no other side effects but I could be mistaken.

**Comment by arokem** [comment]
Oh weird. At some earlier stage, I downloaded the built docs and there was the text "Home" in the middle of the page, but it doesn't seem to be the case now. So, ignore that comment.
